File: dotnet/SOLID/Models/Customer.cs
using dotnet.SOLID.Interface;

namespace dotnet.SOLID.Models;

public class Customer(string name) : IUserManagement, ISubscriptionNotificationService
{
    public string Name { get; set; } = name;
    public string Email { get; set; } = string.Empty;


    public void UpdateUserProfile(Customer customer)
    {
        Console.WriteLine("Updating user profile for " + customer.Name);
    }

    public void ChangePassword(Customer customer, string newPassword)
    {
        Console.WriteLine("Changing password for " + customer.Name);
    }

    public void SubscribeToNewProductsAvailability()
    {
        Console.WriteLine("Subscribed to new products availability");
    }

    public void SubscribeToSmsNotifications()
    {
        Console.WriteLine("Subscribed to SMS notifications");
    }
}

File: dart_lang/lib/a_solid/d_interface_segregation/b_post_refactor/customer.dart
import 'package:dart_lang/a_solid/d_interface_segregation/b_post_refactor/subscription_notification_service.dart';
import 'package:dart_lang/a_solid/d_interface_segregation/b_post_refactor/user_manager.dart';

class Customer implements UserManager, SubscriptionNotificationService {
  String? _name;
  String? _email;

  Customer(String name) : _name = name;

  String? get email => _email;
  String? get name => _name;

  @override
  void changePassword(Customer customer, String newPassword) {
    print("Changing password...");
  }

  @override
  void subscribeToMSNotifications() {
    print("Notifying password...");
  }

  @override
  void subscribeToNewProductsAvailability() {
    print("subscribing...");
  }

  @override
  void updateUserProfile(Customer customer) {
    print("updating...");
  }
}
